Fw/Types/ConstExternalString.hpp

```diff
@@ -0,0 +1,59 @@
+// ======================================================================
+// @file   ConstExternalString.hpp
+// @brief  A string backed by an immutable string literal
+// ======================================================================
+
+#ifndef FW_CONST_EXTERNAL_STRING_HPP
+#define FW_CONST_EXTERNAL_STRING_HPP
+
+#include <Fw/FPrimeBasicTypes.hpp>
+#include <Fw/Types/StringBase.hpp>
+
+namespace Fw {
+
+//! A string backed by an immutable string literal
+class ConstExternalString final : public ConstStringBase {
+  public:
+    // ----------------------------------------------------------------------
+    // Construction and destruction
+    // ----------------------------------------------------------------------
+
+    //! Constructor (uninitialized buffer)
+    ConstExternalString() : ConstStringBase(), m_bufferPtr(nullptr), m_bufferSize(0) {}
+
+    //! Constructor (bufferPtr and bufferSize)
+    ConstExternalString(const char* bufferPtr,                //!< The buffer pointer
+                        ConstStringBase::SizeType bufferSize  //!< The buffer size
+                        )
+        : ConstStringBase(), m_bufferPtr(bufferPtr), m_bufferSize(bufferSize + 1) {}
```

Comment:
```suggestion
        : ConstStringBase(), m_bufferPtr(bufferPtr), m_bufferSize(bufferSize) {}
```
I think we should not add one to the buffer size here.